runtime/onert/api/include/nnfw_experimental.h

```diff
@@ -192,6 +195,7 @@ typedef enum
 
 typedef enum
 {
+  NNFW_TRAIN_OPTIMIZER_UNDEF = -1,
```

Comment:
Q : Why `*_UNDEF` are added to enums?

A :  

`nnfw_train_get_traininfo` api is to get the model file's training info to the user application.
Since we can NOT guarantee that the model file always holds every training information, 
some fields can be `invalid` value. <sub> (missing value is represented as `invalid` in ir/core by default) </sub> value.

So, To represent `invalid` value in the user application, `*_UNDEF` fields become necessary.